### ISSUE TITLE
`persistent-others` doesn't work with relative paths

### ISSUE THREAD

**Issue by kutu-dev** [nix-darwin/nix-darwin#968]
If in the list on `system.defaults.dock.persistent-others` a path is non-absolute as it [is stated in the wiki](https://daiderd.com/nix-darwin/manual/index.html#opt-system.defaults.dock.persistent-others) is will just generate a broken item in the dock.

This generates broken item with a question mark:
```nix
[
  "~/Documents"
]
```

And this works just fine:
```nix
[
  "/Users/user/Documents"
]
```

**Comment by Enzime** [comment]
@rmgpinto can you take a look at this?

**Comment by rmgpinto** [comment]
Can confirm it only works with absolute paths.

**Comment by Samasaur1** [comment]
@rmgpinto do you mean that in the sense that it *currently* only works with absolute paths (i.e. relative paths are intended to work but are currently broken), or that relative paths are intended to be supported?

**Comment by bestlem** [comment]
Note ~/Developer is not a relative path - it is something that a shell (or nix) knows how to expand to an absolute path).

In this case nix should be expanding ~/Documents to /Users/user/Documents

**Comment by kutu-dev** [comment]
> Note ~/Developer is not a relative path - it is something that a shell (or nix) knows how to expand to an absolute path).
> 
> In this case nix should be expanding ~/Documents to /Users/user/Documents

That's true, also do even nixlang have support for expanding the tilde char? It support would be strange being nixlang a pure language, [it's not mentioned in the docs](https://nix.dev/tutorials/nix-language.html#file-system-paths) as far as I can tell. Maybe [the example in the docs](https://daiderd.com/nix-darwin/manual/index.html#opt-system.defaults.dock.persistent-others)  should be changed.

**Comment by emilazy** [comment]
Yeah I feel like this is just a documentation bug; there’s lots of places that don’t support paths like that and I don’t think nix-darwin should be in the business of trying to implement our own ad‐hoc path expansion for this. I’ll review a PR to change the example.

**Comment by lockejan** [comment]
Just to document a workaround. You can pass in "user" as an argument to your config via flakes and then reference it inside your nix-darwin configuration.✌️

https://github.com/lockejan/nix-dotfiles/blob/a66807572aa94def7f7759eb70c1c2ce89a6555e/flake.nix#L74
https://github.com/lockejan/nix-dotfiles/blob/4077fbb5d1ee0a80ef19062e9d63372efeffbca7/darwin/configuration.nix#L163-L167

**Comment by bueti** [comment]
> > Note ~/Developer is not a relative path - it is something that a shell (or nix) knows how to expand to an absolute path).
> > In this case nix should be expanding ~/Documents to /Users/user/Documents
> 
> That's true, also do even nixlang have support for expanding the tilde char? It support would be strange being nixlang a pure language, [it's not mentioned in the docs](https://nix.dev/tutorials/nix-language.html#file-system-paths) as far as I can tell. Maybe [the example in the docs](https://daiderd.com/nix-darwin/manual/index.html#opt-system.defaults.dock.persistent-others) should be changed.

I feel like this used to work before upgrading to Sequoia. The `~` character is mentioned here: https://nix.dev/manual/nix/2.18/language/ (search for `~/.config`).

**Comment by sliminality** [comment]
The example for `persistent-apps` in the manual currently uses tildes, so if they're not working it might be best to change this example?
https://github.com/LnL7/nix-darwin/blob/95eac71bf52b271523d0ca81dbbeb3182990fc24/modules/system/defaults/dock.nix#L199-L202
